Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -1168,6 +1168,31 @@ void PcapLiveDevice::setDefaultGateway()
 #endif
 }
 
+std::vector<IPAddress> PcapLiveDevice::getIPAddresses() const
+{
+	std::vector<IPAddress> results;
+	for (const auto& address : m_Addresses)
+	{
+		in_addr* ipv4Addr = internal::try_sockaddr2in_addr(address.addr);
+		if (ipv4Addr != nullptr)
+		{
+			results.push_back(IPv4Address(ipv4Addr->s_addr));
```

Comment:
Could be `emplace_back`, but it's also fine now.

Reply:
As in this?
```suggestion
			results.emplace_back(ipv4Addr->s_addr);
```

Reply:
I think `push_back(std::move(IPv4Address))` is equal to `emplace_back(IPv4Address`

how about this?
```
results.emplace_back(ipv4Addr->s_addr);
```
But it might be too trivial to distinguish the difference.

Reply:
Tested it on VS. That works for the IPv4, due to the `uint32_t` constructor. The IPv6 gets confused tho coz both v4 and v6 have a constructor from pointer.

Reply:
> But it might be too trivial to distinguish the difference.

Eh, it technically can save a single move constructor if the compiler does not optimize it away anyway.